packages/core/src/utils/surfaceUtils.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
```

Comment:
At a minium, add unit tests for this file and some comments with examples of what these regexs are looking for.

Reply:
This is the existing Personalization `surfaceUtils`. I have extracted it at the core level cause it is needed in Brand Concierge Component too. I will move the tests as well.